File: src/main/java/com/emr/emr_system/modules/medicine/entity/Medicine.java
package com.emr.emr_system.modules.medicine.entity;

import com.emr.emr_system.shared.enums.MedicineUnit;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "medicines")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Medicine {

    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    @Column(name = "category_id")
    private UUID categoryId;

    @Column(name = "code", nullable = false, unique = true, length = 30)
    private String code;

    @Column(name = "name", nullable = false, length = 200)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "unit", nullable = false, length = 20)
    private MedicineUnit unit;

    @Column(name = "manufacturer", length = 200)
    private String manufacturer;

    @Column(name = "description")
    private String description;

    @Column(name = "side_effects")
    private String sideEffects;

    @Column(name = "price", precision = 12, scale = 2)
    private BigDecimal price;

    @Column(name = "stock_quantity", nullable = false)
    private Integer stockQuantity;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/medicine.ts
export interface Medicine {
  id: string;
  categoryId: string;
  categoryName?: string;
  code: string;
  name: string;
  unit: MedicineUnit;
  manufacturer?: string;
  description?: string;
  sideEffects?: string;
  price: number;
  stockQuantity: number;
  isActive: boolean;
  createdAt?: string;
  updatedAt?: string;
}

export enum MedicineUnit {
  TABLET = "TABLET",
  CAPSULE = "CAPSULE",
  SYRUP = "SYRUP",
  ML = "ML",
  MG = "MG",
  VIAL = "VIAL",
  TUBE = "TUBE",
  PACK = "PACK",
  BOX = "BOX"
}

export interface MedicineCategory {
  id: string;
  name: string;
  description?: string;
  createdAt?: string;
}

export interface MedicineCreateRequest {
  categoryId: string;
  code: string;
  name: string;
  unit: MedicineUnit | undefined;
  manufacturer?: string;
  description?: string;
  sideEffects?: string;
  price: number;
  stockQuantity?: number;
  isActive?: boolean;
}

export interface MedicineUpdateRequest {
  categoryId: string;
  code: string;
  name: string;
  unit: MedicineUnit;
  manufacturer?: string;
  description?: string;
  sideEffects?: string;
  price: number;
}

export interface MedicineStatusRequest {
  isActive: boolean;
}

export interface MedicineStockRequest {
  quantity: number;
}